### PR TITLE
Basic speech parsers

### PR DESCRIPTION
**Pull request by Pokeizel**
Adds basic parsers for getting the speaker, speaker data, and speech from a chatterbox's content.
New functions: 

- ChatterboxGetSpeaker( chatterbox, contentIndex)
- ChatterboxGetSpeakerData( chatterbox, contentIndex)
- ChatterboxGetSpeech( chatterbox, contentIndex)

Added macros:

- CHATTERBOX_SPEAKERDATA_TAG_OPEN
- CHATTERBOX_SPEAKERDATA_TAG_CLOSE

### CODE REVIEW

**Comment by JujuAdams** [comment]
Hiya, thanks for this PR. Couple of requests:

1. Please use spaces not tabs (4 spaces per tab, there or thereabouts)
2. Please extend the whitespace removal such that `name:      speech` trims off all whitespace between `:` and `speech`
3. Please handle cases where there is no `:` in the string

**Comment by JujuAdams** [comment]
(This also partially addresses issue #36)

**Comment by JujuAdams** [comment]
Another thing occurs to me too:
4. Error detection for when `CHATTERBOX_SPEAKERDATA_TAG_OPEN` is detected and not `CHATTERBOX_SPEAKERDATA_TAG_CLOSE`, and vice versa

**Comment by Pokeizel** [comment]
> Another thing occurs to me too: 4. Error detection for when `CHATTERBOX_SPEAKERDATA_TAG_OPEN` is detected and not `CHATTERBOX_SPEAKERDATA_TAG_CLOSE`, and vice versa
I just saw this while pushing a commit, I just tested that and nothing crazy happens, but it could be cleaned up a little. Not now though as I should've been asleep like 5 hours ago

File: scripts/ChatterboxGetSpeech/ChatterboxGetSpeech.gml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    var _speech = _str;
+    if (_str != undefined)
+    {
+        var _colon = string_pos(":", _str);
+        var _space = 0;
+        if (_colon != 0) //If there *is* a colon
+        {
+            _speech = string_delete(_str, 1, _colon); //Remove the colon and all behind it,
+        }
+        if (string_pos(" ", _speech) == 1) //If there's as the first character
```
Comment (JujuAdams) [inline scripts/ChatterboxGetSpeech/ChatterboxGetSpeech.gml:18]: Chatterbox automatically strips leading whitespace from every line during compilation. All string manipulation should only happen if there's a colon.

File: scripts/ChatterboxGetSpeakerData/ChatterboxGetSpeakerData.gml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+    var _str = ChatterboxGetContent(_chatterbox, _index);
+    if (_str != undefined)
+    {
+        var _colon = string_pos(":", _str);
+        var _c1 = CHATTERBOX_SPEAKERDATA_TAG_OPEN;
+        var _c2 = CHATTERBOX_SPEAKERDATA_TAG_CLOSE;
+        var _char_l = string_pos(_c1, _str);
+        var _char_r = string_pos(_c2, _str);
+        if (_char_l and _char_r < _colon)
```
Comment (JujuAdams) [inline scripts/ChatterboxGetSpeakerData/ChatterboxGetSpeakerData.gml:18]: Implicitly casting from a number to a bool is going to start throwing warnings in future versions of GameMaker. Please make this check explicit.

File: scripts/ChatterboxGetSpeakerData/ChatterboxGetSpeakerData.gml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+/// @param chatterbox
+/// @param contentIndex
+/// @param [default]
+
+function ChatterboxGetSpeakerData(_chatterbox, _index, _default = undefined)
+{
+    var _str = ChatterboxGetContent(_chatterbox, _index);
+    if (_str != undefined)
+    {
+        var _colon = string_pos(":", _str);
```
Comment (JujuAdams) [inline scripts/ChatterboxGetSpeakerData/ChatterboxGetSpeakerData.gml:13]: Please make the colon presence check explicit (i.e. not `_char_r < _colon`) as this is clearer. Less clever, but clearer.

File: scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+function ChatterboxGetSpeaker(_chatterbox, _index)
+{
+    var _str = ChatterboxGetContent(_chatterbox, _index);
+    if (_str != undefined)
+    {
+        var _colon = string_pos(":", _str);
+        if (!_colon)
+        {
+            show_debug_message("Chatterbox: No speaker found.") return "";
```
Comment (JujuAdams) [inline scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml:14]: Please use `__ChatterboxTrace()` to output messages. Additionally, I'm unsure if it's more suitable to return `""` or `undefined` here. Thoughts?

File: scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        }
+        var _name = string_copy(_str, 1, _colon-1);
+        var _c1 = CHATTERBOX_SPEAKERDATA_TAG_OPEN;
+        var _c2 = CHATTERBOX_SPEAKERDATA_TAG_CLOSE;
+        var _char_l = string_pos(_c1, _str);
+        var _char_r = string_pos(_c2, _str);
+        if (_char_l and _char_r < _colon) //If there are tags before the colon
```
Comment (JujuAdams) [inline scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml:21]: Implicitly casting from a number to a bool is going to start throwing warnings in future versions of GameMaker. Please make this check explicit.

File: scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml (hunk 4 above)
Comment (Pokeizel) [reply scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml:14]: I think some people might want certain lines of text with an empty speaker, and returning `undefined` might force them to either add a single colon at the start of a line, which looks a bit ugly and wouldn't make much sense for some writers in my opinion, or code it so if the string is `"undefined"`, the speaker and/or namebox isn't drawn, but they could already do that with just `""`

File: scripts/ChatterboxGetSpeakerData/ChatterboxGetSpeakerData.gml (hunk 3 above)
Comment (Pokeizel) [reply scripts/ChatterboxGetSpeakerData/ChatterboxGetSpeakerData.gml:13]: Could you explain further? I'm not sure what you mean with explicit. I changed it to `if (_tag_l != 0 and _tag_r < _colon_pos)`, but I'm not sure if that's what you meant

**Comment by Pokeizel** [comment]
I didn't mean to reference Issue number 2 on the latest commit. My bad.

File: scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml (hunk 4 above)
Comment (JujuAdams) [reply scripts/ChatterboxGetSpeaker/ChatterboxGetSpeaker.gml:14]: `""` works for me then

**Pull request merged**